log/logClient.go

```diff
@@ -271,8 +271,11 @@ func watchAlertsHelper(res *pb.Alert, o Options) error {
 
 	str := ""
 
-	if o.JSON {
+	if o.JSON || o.EventChan != nil {
 		arr, _ := json.Marshal(res)
+		if o.EventChan != nil {
+			o.EventChan <- arr
```

Comment:
Are you saying it would be better to send `res` as in `o.EventChan <- res` so that we can directly access the structure without having to unmarshal?

Two reasons:
1. Alerts and Telemetry have different protobufs ... Wanted EventChan to accept something generic.
2. would require addition kubearmor-client protobuf dependency ... we would run into problem where we have to update protobuf dependency everytime change is done to protobuf. Not a big deal though.

Reply:
Perhaps let's keep it this way only, It may introduce further complexity if we do it other way around.

Reply:
changed to using agnostic channels. That makes more sense. Thanks